test/src/main/java/org/apache/accumulo/test/metrics/fate/ZooMutatorTest.java

```diff
@@ -75,66 +70,65 @@
-      ConcurrentHashMap<Integer,Integer> countCounts = new ConcurrentHashMap<>();
+    // This map is used to ensure multiple threads do not successfully write the same value and no
+    // values are skipped. The hash in the value also verifies similar things in a different way.
+    ConcurrentHashMap<Integer,Integer> countCounts = new ConcurrentHashMap<>();
 
-      for (int i = 0; i < 16; i++) {
-        futures.add(executor.submit(() -> {
-          try {
+    for (int i = 0; i < 16; i++) {
+      futures.add(executor.submit(() -> {
+        try {
 
-            int count = -1;
-            while (count < 200) {
-              byte[] val =
-                  zk.mutateOrCreate("/test-zm", initialData.getBytes(UTF_8), this::nextValue);
-              int nextCount = getCount(val);
-              assertTrue("nextCount <= count " + nextCount + " " + count, nextCount > count);
-              count = nextCount;
-              countCounts.merge(count, 1, Integer::sum);
-            }
-
-          } catch (Exception e) {
-            throw new RuntimeException(e);
+          int count = -1;
+          while (count < 200) {
+            byte[] val =
+                zk.mutateOrCreate("/test-zm", initialData.getBytes(UTF_8), this::nextValue);
+            int nextCount = getCount(val);
+            assertTrue("nextCount <= count " + nextCount + " " + count, nextCount > count);
+            count = nextCount;
+            countCounts.merge(count, 1, Integer::sum);
           }
-        }));
-      }
 
-      // wait and check for errors in background threads
-      for (Future<?> future : futures) {
-        future.get();
-      }
-      executor.shutdown();
+        } catch (Exception e) {
+          throw new RuntimeException(e);
+        }
+      }));
+    }
 
-      byte[] actual = zk.getData("/test-zm");
-      int settledCount = getCount(actual);
+    // wait and check for errors in background threads
+    for (Future<?> future : futures) {
+      future.get();
+    }
+    executor.shutdown();
 
-      assertTrue(settledCount >= 200);
+    byte[] actual = zk.getData("/test-zm");
+    int settledCount = getCount(actual);
 
-      String expected = initialData;
+    assertTrue(settledCount >= 200);
 
-      assertEquals(1, (int) countCounts.get(0));
+    String expected = initialData;
 
-      for (int i = 1; i <= settledCount; i++) {
-        assertEquals(1, (int) countCounts.get(i));
-        expected = nextValue(expected);
-      }
+    assertEquals(1, (int) countCounts.get(0));
 
-      assertEquals(settledCount + 1, countCounts.size());
-      assertEquals(expected, new String(actual, UTF_8));
+    for (int i = 1; i <= settledCount; i++) {
+      assertEquals(1, (int) countCounts.get(i));
+      expected = nextValue(expected);
     }
+
+    assertEquals(settledCount + 1, countCounts.size());
+    assertEquals(expected, new String(actual, UTF_8));
+
+    szk.close();
```

Comment:
Instead of a close here, the szk should be declared as a resource in a try-with-resources block, so it gets closed, even if the other code throws an exception.

Reply:
ZooKeeperTestingServer does not implement auto-closable so it wont work in a regular try-with-resources block. I could either make the ZooKeeperTestingServer a member variable and use a try-finally to close it or, have a @BeforeClass and @AfterClass to initialize and then close it. Not sure which would be best in this case.

Reply:
It has a close method, so it might as well implement the AutoCloseable interface. Whether you use the before/after class methods or try-with-resources or try-finally pattern, might depend on case-by-case basis, but there's no reason not to add the interface to the class, since it already has an implementing method.

Reply:
These changes are included in the most recent commit.